web-client/public/js/update-app.js

```diff
@@ -322,6 +328,34 @@ const hasExpressionData = (sheets) => {
     return false;
 };
 
+const updateSpeciesMenu = () => {
+    $(SPECIES_DISPLAY).val(grnState.genePageData.species);
+    $(SPECIES_BUTTON_CRESS + " span").removeClass("glyphicon-ok");
+    $(SPECIES_BUTTON_FLY + " span").removeClass("glyphicon-ok");
+    $(SPECIES_BUTTON_HUMAN + " span").removeClass("glyphicon-ok");
+    $(SPECIES_BUTTON_YEAST + " span").removeClass("glyphicon-ok");
+    $(SPECIES_BUTTON_NEMATODE + " span").removeClass("glyphicon-ok");
+    $(SPECIES_BUTTON_MOUSE + " span").removeClass("glyphicon-ok");
+    if ($(SPECIES_DISPLAY).val() === "Arabidopsis_thaliana") {
+        $(SPECIES_BUTTON_CRESS + " span").addClass("glyphicon-ok");
+    }
+    if ($(SPECIES_DISPLAY).val() === "Drosophila_melanogaster") {
+        $(SPECIES_BUTTON_FLY + " span").addClass("glyphicon-ok");
+    }
+    if ($(SPECIES_DISPLAY).val() === "Caenorhabditis_elegans") {
+        $(SPECIES_BUTTON_NEMATODE + " span").addClass("glyphicon-ok");
+    }
+    if ($(SPECIES_DISPLAY).val() === "Homo_sapiens") {
+        $(SPECIES_BUTTON_HUMAN + " span").addClass("glyphicon-ok");
+    }
+    if ($(SPECIES_DISPLAY).val() === "Mus_musculus") {
+        $(SPECIES_BUTTON_MOUSE + " span").addClass("glyphicon-ok");
+    }
+    if ($(SPECIES_DISPLAY).val() === "Saccharomyces_cerevisiae") {
+        $(SPECIES_BUTTON_YEAST + " span").addClass("glyphicon-ok");
+    }
+};
```

Comment:
Some consolidation potential here also, but not quite as pronounced as the other one. Again, it can wait.